src/server/game/Entities/Creature/CreatureData.h

```diff
@@ -495,6 +495,7 @@ struct TC_GAME_API CreatureTemplate
     float   BaseVariance;
     float   RangeVariance;
     uint32  unit_class;                                     // enum Classes. Note only 4 classes are known for creatures.
+    bool    IsEnergyUser;
```

Comment:
I don't like that this is only for energy, why not making it a bool/flag "show all power types", not special casing energy but rather just using default power type for class (db2) (that would also be able to work for warriors)

Reply:
Does that not mean that most creatures under unit_class 1 (warrior) will regenerate fury if this bool happens to be true? The only creatures that would resemble that are related to ImitationAI from BfA Islands or Arathi Basin's bots when the brawl is up.

Reply:
regenerate automatically? obviously no, it will just work with standard mechanics

Reply:
But what's the point of doing it for both if UNIT_CLASS_WARRIOR will not be able to regenerate fury when attacking?